endaq/calc/stats.py

```diff
@@ -6,6 +6,342 @@
 
 import numpy as np
 import pandas as pd
+from scipy import signal
+import plotly.graph_objects as go
+
+from endaq.plot import rolling_min_max_envelope
+from endaq.calc import filters, integrate, utils, shock
+
+
+def shock_vibe_metrics(
+        df: pd.DataFrame,
+        tukey_percent: float = 0.1,
+        highpass_cutoff: float = None,
+        vel_disp_multiplier: float = 1.0,
+        zero: typing.Literal["start", "mean", "median", None] = "median",
+        include_integration: bool = True,
+        include_pseudo_velocity: bool = True,
+        damp: float = 0.05,
+        init_freq: float = 1.0,
+        bins_per_octave: float = 12,
+        include_resultant: bool = False,
+        display_plots: bool = False,
+) -> pd.DataFrame:
+    """
+    Compute the following shock and vibration metrics for a given time series dataframe
+        - Peak Absolute Acceleration
+        - RMS Acceleration
+        - Peak Absolute Velocity
+        - RMS Velocity
+        - Peak Absolute Displacement
+        - RMS Displacement
+        - Peak Pseudo Velocity
+    :param df: the input dataframe with an index defining the time in seconds or datetime
+    :param tukey_percent: the portion of the time series to apply a Tukey window (a taper that forces beginning and end to 0), default is 0.1
+        - Note that the RMS metrics will only be computed on the portion of time that isn't tapered
+    :param zero: the output quantity driven to zero prior to the calculations
+        `None` does nothing
+        `"start"` forces the first datapoint to 0,
+        `"mean"` chooses ``-np.mean(output)`` & `"median"` (default) chooses
+        ``-np.median(output)``
```

Comment:
`zero` is a bit too ambiguous.  I'd suggest something more descriptive.  Off the top of my head, `offset` or `offset_normalization`.

Reply:
I copied this from `endaq.calc.integrate` that uses this parameter and options so if we change it in this function we'd have to change it there and that function has been out for some time. So I think it's best to keep as is?

Reply:
Yeah, if that's the case let's not change that in this PR.